app/helpers/training_helper.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+#= Helpers for training views
+module TrainingHelper
+  # Given a module slug, it returns the first library that has a
+  # category including that module slug. It returns nil if no such
+  # library is found.
+  def find_library_from_module_slug(module_slug)
+    TrainingLibrary.all.find_each do |library|
+      library.categories.each do |category|
+        next unless category.key?('modules')
+        category['modules'].each do |mod|
+          return library if mod['slug'] == module_slug
+        end
+      end
+    end
+  end
+
+  # Given a slide slug, it returns the first module including it.
```

Comment:
As a TrainingSlide instance method, you could implement this as 

```
TrainingModule.all.detect { |tm| tm.slide_slugs.include? slug }
```

Reply:
Much more idiomatic, I like it!